### PR TITLE
Clean up of the index utility code

### PR DESCRIPTION
**Pull request by apavlo**
This is to help make it easier for the students to debug their project in the class. I am going to merge this unless anybody complains.

1. Added GetInfo methods for TupleKey and GenericKey.
2. Added IndexUtil::GetInfo(ItemPointer)
3. Added examples of debug output in BwTreeIndex::InsertEntry + DeleteEntry

https://www.youtube.com/watch?v=_QXRQRoO4IE

### CODE REVIEW

File: src/include/index/compact_ints_key.h [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
+//
+// compact_ints_key.h
+//
+// Identification: src/include/index/compact_ints_key.h
+//
+// Copyright (c) 2015-2018, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
 //===----------------------------------------------------------------------===//
```
Comment (pmenon) [inline src/include/index/compact_ints_key.h]: Duplicate copyright.

File: src/include/index/generic_key.h [hunk 2]
```diff
@@ -1,3 +1,15 @@
...
+//
+// generic_key.h
+//
+// Identification: src/include/index/generic_key.h
+//
+// Copyright (c) 2015-2018, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
 //===----------------------------------------------------------------------===//
```
Comment (pmenon) [inline src/include/index/generic_key.h]: Duplicate copyright.

File: src/include/index/tuple_key.h [hunk 3]
```diff
@@ -1,3 +1,15 @@
...
+// tuple_key.h
+//
+// Identification: src/include/index/tuple_key.h
+//
+// Copyright (c) 2015-2018, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
 
 //===----------------------------------------------------------------------===//
```
Comment (pmenon) [inline src/include/index/tuple_key.h]: Duplicate copyright.

File: src/index/bwtree_index.cpp [hunk 4]
```diff
@@ -1,3 +1,15 @@
...
+//
+// bwtree_index.cpp
+//
+// Identification: src/index/bwtree_index.cpp
+//
+// Copyright (c) 2015-2018, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
 //===----------------------------------------------------------------------===//
```
Comment (pmenon) [inline src/index/bwtree_index.cpp]: Duplicate copyright.

**Comment by apavlo** [comment]
@pmenon Thanks! I will merge when the build passes.

**Pull request merged**